### ISSUE TITLE
--expand-errors blow-up behavior for conjunct chains

### ISSUE THREAD

**Issue by tjhance** [verus-lang/verus#400]
I'm seeing behavior where running --expand error on a conjunct-list gives AIR which is ... I think cubic in the # of conjuncts, though I'm not entirely sure. I'm pretty sure it's more than quadratic, though, and it blows up to 100,000s of lines for reasonable programs.

For example, for a simple conjunct like this:

```rust
spec fn boolfn(a: bool, b: bool, c: bool, d: bool, e: bool) -> bool {
    &&& a
    &&& b
    &&& c
    &&& d
    &&& e
}
```

We could generate AIR that looks like:

```
(assert a)
(assume a)
(assert b)
(assume b)
(assert c)
(assume c)
(assert d)
(assume d)
(assert e)
```

Instead we currently make:

```rust
(assert
   ("split postcondition failure" "from this function call")
   (%B (B a~2@))
  )
  (assert
   ("split postcondition failure" "from this function call")
   (=>
    (%B (B a~2@))
    (=>
     (%B (B a~2@))
     (=>
      (%B (B a~2@))
      (=>
       (%B (B a~2@))
       (%B (B b~4@))
  )))))
  (assert
   ("split postcondition failure" "from this function call")
   (=>
    (=>
     (%B (B a~2@))
     (=>
      (%B (B a~2@))
      (=>
       (%B (B a~2@))
       (=>
        (%B (B a~2@))
        (%B (B b~4@))
    ))))
    (=>
     (=>
      (%B (B a~2@))
      (=>
       (%B (B a~2@))
       (=>
        (%B (B a~2@))
        (%B (B b~4@))
     )))
     (=>
      (and
       (%B (B a~2@))
       (%B (B b~4@))
      )
      (%B (B c~6@))
  ))))
  (assert
   ("split postcondition failure" "from this function call")
   (=>
    (=>
     (=>
      (%B (B a~2@))
      (=>
       (%B (B a~2@))
       (=>
        (%B (B a~2@))
        (=>
         (%B (B a~2@))
         (%B (B b~4@))
     ))))
     (=>
      (=>
       (%B (B a~2@))
       (=>
        (%B (B a~2@))
        (=>
         (%B (B a~2@))
         (%B (B b~4@))
      )))
      (=>
       (and
        (%B (B a~2@))
        (%B (B b~4@))
       )
       (%B (B c~6@))
    )))
    (=>
     (and
      (and
       (%B (B a~2@))
       (%B (B b~4@))
      )
      (%B (B c~6@))
     )
     (%B (B d~8@))
  )))
  (assert
   ("split postcondition failure" "from this function call")
   (=>
    (and
     (and
      (and
       (%B (B a~2@))
       (%B (B b~4@))
      )
      (%B (B c~6@))
     )
     (%B (B d~8@))
    )
    (%B (B e~10@))
```

Note that there are a bunch of redundant implication chains like `(a => a => a => a => b)`.

**Comment by tjhance** [comment]
This was fixed partially in #409 - further improvements should be possible as I detailed above, although they may be low priority.

The generated AIR currently looks like this:

```
(check-valid
 (declare-const a~2@ Bool)
 (declare-const b~4@ Bool)
 (declare-const c~6@ Bool)
 (declare-const d~8@ Bool)
 (declare-const e~10@ Bool)
 (axiom fuel_defaults)
 (block
  (assert
   ("split postcondition failure" "from this function call")
   (%B (B a~2@))
  )
  (assert
   ("split postcondition failure" "from this function call")
   (=>
    (%B (B a~2@))
    (%B (B b~4@))
  ))
  (assert
   ("split postcondition failure" "from this function call")
   (=>
    (and
     (%B (B a~2@))
     (%B (B b~4@))
    )
    (%B (B c~6@))
  ))
  (assert
   ("split postcondition failure" "from this function call")
   (=>
    (and
     (and
      (%B (B a~2@))
      (%B (B b~4@))
     )
     (%B (B c~6@))
    )
    (%B (B d~8@))
  ))
  (assert
   ("split postcondition failure" "from this function call")
   (=>
    (and
     (and
      (and
       (%B (B a~2@))
       (%B (B b~4@))
      )
      (%B (B c~6@))
     )
     (%B (B d~8@))
    )
    (%B (B e~10@))
))))
```

**Comment by utaal** [comment]
[Verus all-hands]

@tjhance fixed this.
There may be opportunities to make the AIR encoding more compact, but it doesn't seem to have been an issue so far.
